homeassistant/components/jellyfin/media_source.py

```diff
@@ -324,11 +324,14 @@ def _get_audio_stream_url(self, media_item: dict[str, Any]) -> str:
 
 def _media_mime_type(media_item: dict[str, Any]) -> str:
     """Return the mime type of a media item."""
-    media_source = media_item[ITEM_KEY_MEDIA_SOURCES][0]
-    path = media_source[MEDIA_SOURCE_KEY_PATH]
-    mime_type, _ = mimetypes.guess_type(path)
+    if media_item[ITEM_KEY_MEDIA_SOURCES]:
+        media_source = media_item[ITEM_KEY_MEDIA_SOURCES][0]
+        path = media_source[MEDIA_SOURCE_KEY_PATH]
+        mime_type, _ = mimetypes.guess_type(path)
 
-    if mime_type is not None:
-        return mime_type
+        if mime_type is not None:
+            return mime_type
 
-    raise BrowseError(f"Unable to determine mime type for path {path}")
+        raise BrowseError(f"Unable to determine mime type for path {path}")
+
+    raise BrowseError("Unable to determine mime type for item without media source")
```

Comment:
If the missing key leads to an exception, couldn't that be done at the start?  Then you probably don't need to indent so much of the other code.

Reply:
Agreed, that is indeed a nicer way of writing this up. Done!